Remotion/Data/DomainObjects.UnitTests/Linq/IntegrationTests/ObjectIDIntegrationTest.cs

```diff
@@ -255,8 +255,9 @@ where possibleItems.Contains(o)
           select o;
       Assert.That(
           () => CheckQueryResult(orders, DomainObjectIDs.Order1, DomainObjectIDs.Order3),
-          Throws.InstanceOf<NotSupportedException>()
-              .With.Message.EqualTo("Objects of type 'Remotion.Data.DomainObjects.UnitTests.TestDomain.Order_AssembledTypeProxy_1' cannot be used as data parameter value."));
+          Throws.InstanceOf<InvalidOperationException>().With.Message.EqualTo(
+              "The parameter value's type 'Remotion.Data.DomainObjects.DomainObject[]' cannot be mapped by "
+              + "Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters.SimpleTypeQueryParameterRecordDefinitionFinder."));
```

Comment:
getting an invalidopex instead of a notsupportedex here in the itnegration test feels like a regression. there is also a not very helpful bit of information with an internal type. I would prefer the original message or something close to it, e.g. `A collection of objects of type '<type>' cannot be used as a data parameter value. Only collections of simple values are supported.` (not sure whether to use use 'simple values' or 'scalar values' in the phrasing.

Reply:
I am happy to switch back to NotSupportedException, but if I recall correctly, we talked about this.
As for the internal class: this is an extension point; the exception is only thrown when the developer tries to use a parameter value that is a collection of objects whose type has no corresponding `TableTypeDefinition`; the exception is supposed to help the developer realize that he needs to decorate `SimpleTypeQueryParameterRecordDefinitionFinder` with a class that maps the item type to a `TableTypeDefinition`.